libs/features/mas/src/price/template.js

```diff
@@ -103,17 +103,44 @@ export const renderSpan = (
     );
 };
 
+function encodeLinks(literal) {
+  literal = literal.replaceAll('</a>', '&lt;/a&gt;');
+
+  const regex = /<a [^>]+(>|$)/g;
+  const matches = literal.match(regex);
+  matches?.forEach((match) => {
+    const encodedMatch = match.replace('<a ', '&lt;a ').replace('>', '&gt;');
+    literal = literal.replaceAll(match, encodedMatch);
+  });
+
+  return literal;
+}
+
+function decodeLinks(literal) {
+  literal = literal.replaceAll('&lt;/a&gt;', '</a>');
+
+  const regex = /&lt;a (?!&gt;)(.*?)(&gt;|$)/g;
+  const matches = literal.match(regex);
+  matches?.forEach((match) => {
+    const encodedMatch = match.replace('&lt;a ', '<a ').replace('&gt;', '>');
+    literal = literal.replaceAll(match, encodedMatch);
+  });
+
+  return literal;
+}
+
 export function formatLiteral(literals, locale, key, parameters) {
     const literal = literals[key];
     if (literal == undefined) {
         /* c8 ignore next 2 */
         return '';
     }
     try {
-        return new IntlMessageFormat(
-            literal.replace(htmlPattern, ''),
+        const formattedLiteral = new IntlMessageFormat(
+            encodeLinks(literal).replace(htmlPattern, ''),
```

Comment:
this method call is going to have a small perf impact on all price literals.

Reply:
@Axelcureno done. Please check again.